test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,115 @@
+#include "test/common/upstream/load_balancer_fuzz_base.h"
+
+#include "test/common/upstream/utility.h"
+
+namespace Envoy {
+
+namespace Random {
+uint64_t FakeRandomGenerator::random() {
+  uint8_t index_of_data = counter % bytestring_.size();
+  ++counter;
+  ENVOY_LOG_MISC(trace, "random() returned: {}", bytestring_.at(index_of_data));
+  return bytestring_.at(index_of_data);
+}
+} // namespace Random
+
+namespace Upstream {
+
+// Anonymous namespace for helper functions
+namespace {
+std::vector<uint64_t>
+constructByteVectorForRandom(const Protobuf::RepeatedField<uint64_t>& byteString) {
+  std::vector<uint64_t> byteVector;
```

Comment:
You should preallocate space for this by calling [std::vector::reserve()](https://en.cppreference.com/w/cpp/container/vector/reserve) before entering the loop. That way the vector doesn't have to keep growing its internal storage periodically.

Reply:
Clang tidy caught this, fixed it in commit to #13400